examples/algorithm_progression/xarray_sweep.py

```diff
@@ -0,0 +1,87 @@
+from pathlib import Path
+
+import numpy as np
+import xarray as xr
+from algorithm_sweep import AlgorithmSweep
+
+import large_image
+
+
+class AlgorithmSweepXArray(AlgorithmSweep):
+    def create_ds(self):
+        dim_sizes = dict(
+            **{k: len(v) for k, v in self.input_params.items()},
+            **{
+                'y': self.source_metadata['sizeY'],
+                'x': self.source_metadata['sizeX'],
+                'c': 4,
+            },
+        )
+
+        ds = xr.Dataset(
+            {
+                'results': (
+                    list(dim_sizes.keys()),
+                    np.zeros(
+                        tuple(dim_sizes.values()),
+                    ),
+                )
+            },
+            coords=self.input_params,
+            attrs={
+                'name': f'{self.algorithm_name} iterative results',
+                'description': f'{self.algorithm_name} performed on {self.input_filename}',
+            },
+        )
+        return ds
+
+    def initialize_storage(self):
+        self.source = large_image.open(self.input_filename)
+        self.source_metadata = self.source.getMetadata()
+        self.zarr_path = Path(self.output_dir, 'results.zarr')
+        ds = self.create_ds()
+        ds.to_zarr(self.zarr_path, mode='w', compute=False)
+
+    def apply_algorithm(self, param_combo):
+        current_coords = {
+            k: [param_combo[i]] for i, k in enumerate(self.input_params.keys())
+        }
+        output_data = np.empty(
+            (
```

Comment:
We really don't want to allocate an array the size of our image -- some don't fit into memory.

Reply:
Changed in [980bc50](https://github.com/girder/large_image/pull/1077/commits/980bc50d5ca759524577d1656b72dfa62f6aab96)